core/src/main/scala/org/bitcoins/core/protocol/dlc/DLCTemplate.scala

```diff
@@ -0,0 +1,296 @@
+package org.bitcoins.core.protocol.dlc
+
+import org.bitcoins.core.currency._
+import org.bitcoins.core.protocol.dlc.ContractOraclePair._
+import org.bitcoins.core.protocol.dlc.DLCTemplate.validateMatchingOracleDescriptors
+import org.bitcoins.core.protocol.tlv._
+
+sealed trait DLCTemplate {
+  def totalCollateral: CurrencyUnit
+
+  def oracles: Vector[OracleAnnouncementTLV]
```

Comment:
this can be defined in terms of `oracleInfo`

Reply:
I think it's better to do it this way so users can enter in each announcement themselves